common/frame.go

```diff
@@ -592,23 +591,56 @@ func (f *Frame) waitForFunction(
 
 	pageFn := `
 		(injected, predicate, polling, timeout, ...args) => {
-			return injected.waitForPredicateFunction(predicate, polling, timeout, ...args);
+			const fn = (...args) => {
+				return predicate(...args) === true || injected.continuePolling;
```

Comment:
This is wrong: the promise should resolve on a truthy value, not only `true`. See the [Puppeteer docs](https://puppeteer.github.io/puppeteer/docs/puppeteer.page.waitforfunction/).

The resolved value should also be a `JSHandle` if possible.

Reply:
Resolved by b0ac4d8.